core/akka-runtime/src/main/scala/com/ing/baker/runtime/akka/actor/process_index/ProcessIndex.scala

```diff
@@ -97,9 +104,11 @@ class ProcessIndex(recipeInstanceIdleTimeout: Option[FiniteDuration],
 
   def updateCache() = {
     // TODO this is a synchronous ask on an actor which is considered bad practice, alternative?
-    val futureResult = recipeManager.ask(GetAllRecipes)(updateCacheTimeout).mapTo[AllRecipes]
-    val allRecipes = Await.result(futureResult, updateCacheTimeout)
-    recipeCache ++= allRecipes.recipes.map(ri => ri.compiledRecipe.recipeId -> (ri.compiledRecipe, ri.timestamp))
+    val futureResult: Future[Seq[(CompiledRecipe, Long)]] = recipeManager.all()
+    val allRecipes: Seq[(CompiledRecipe, Long)] = Await.result(futureResult, updateCacheTimeout)
```

Comment:
just wonder why do we need an `Await` here, can't we just do recipeManager.all map { allRecipes => ... update cache here ... } ?

Reply:
The problem with this will be in other place - 
```
def getRecipeWithTimeStamp(recipeId: String): Option[(CompiledRecipe, Long)] =
    recipeCache.get(recipeId) match {
      case None =>
        updateCache()
        recipeCache.get(recipeId)
      case other => other
    }
```

expects updateCache fetches the recipes and returns required instances. If refactor the chain to Future then `createProcessActor` should return Future[ActorRef] that is more complicated than have single await executed once per Node lifecycle